src/nunit.analyzers/ConstActualValueUsage/ConstActualValueUsageAnalyzer.cs

```diff
@@ -24,21 +26,91 @@ public class ConstActualValueUsageAnalyzer : BaseAssertionAnalyzer
         protected override void AnalyzeAssertInvocation(SyntaxNodeAnalysisContext context,
             InvocationExpressionSyntax assertExpression, IMethodSymbol methodSymbol)
         {
+            bool IsLiteralExpression(ExpressionSyntax expression)
+            {
+                if (expression is LiteralExpressionSyntax)
+                    return true;
+
+                if (expression is PrefixUnaryExpressionSyntax prefixUnaryExpression)
+                    return IsLiteralExpression(prefixUnaryExpression.Operand);
+
+                if (expression is BinaryExpressionSyntax binaryExpression)
+                    return IsLiteralExpression(binaryExpression.Left) && IsLiteralExpression(binaryExpression.Right);
+
+                if (expression is ParenthesizedExpressionSyntax parenthesizedExpression)
+                    return IsLiteralExpression(parenthesizedExpression.Expression);
+
+                return false;
+            }
+
+            bool IsConstant(ExpressionSyntax expression)
+            {
+                if (IsLiteralExpression(expression))
+                    return true;
+
+                var argumentSymbol = context.SemanticModel.GetSymbolInfo(expression).Symbol;
+
+                return (argumentSymbol is ILocalSymbol localSymbol && localSymbol.IsConst) ||
+                       (argumentSymbol is IFieldSymbol fieldSymbol && fieldSymbol.IsConst);
+            }
+
+            bool IsEmpty(ExpressionSyntax expression)
+            {
+                return expression is MemberAccessExpressionSyntax memberAccessExpression &&
+                    memberAccessExpression.Name.Identifier.Text == "Empty";
```

Comment:
Personally I don't mind considering `string.Empty` as const for this analyzer, but I'm not sure it applies to any `Empty` property possible.
I would limit it by strings only.

Reply:
I can live with that.